src/app/toolbar/toolbar.component.spec.ts

```diff
@@ -1,22 +1,44 @@
-import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { CardsComponent } from './../cards/cards.component';
+import {
+  ComponentFixture,
+  fakeAsync,
+  TestBed,
+  tick,
+} from '@angular/core/testing';
 
+import { Location } from '@angular/common';
+import { RouterTestingModule } from '@angular/router/testing';
 import { ToolbarComponent } from './toolbar.component';
+import { Router } from '@angular/router';
+import { routes } from '../app-routing.module';
 
 describe('ToolbarComponent', () => {
-  let component: ToolbarComponent;
+  let router: Router;
+  let location: Location;
   let fixture: ComponentFixture<ToolbarComponent>;
 
-  beforeEach(async () => {
-    await TestBed.configureTestingModule({
-      declarations: [ToolbarComponent],
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [RouterTestingModule.withRoutes(routes)],
+      declarations: [ToolbarComponent, CardsComponent],
     }).compileComponents();
 
+    router = TestBed.inject(Router);
+    location = TestBed.inject(Location);
+
     fixture = TestBed.createComponent(ToolbarComponent);
-    component = fixture.componentInstance;
-    fixture.detectChanges();
+    router.initialNavigation();
   });
 
-  it('should create', () => {
-    expect(component).toBeTruthy();
+  it('should render "HOME" link in the toolbar', () => {
+    const link =
+      fixture.debugElement.nativeElement.querySelector('#homeLink').textContent;
+    expect(link).toContain('HOME');
   });
+
+  it('should click in "Home" link and navigate to home page', fakeAsync(() => {
```

Comment:
What's this test doing ? 
you are forcing the router to navigate to home and checking if the path is home :D ? If you want to test the click you would need to spy on the router and listen to it's navigate method to be called with '/' when you click the button :) This type of test would be beneficial if you had some additional url-params.

Reply:
Good to know, thanks. We have recently the discussion that the click and the navigation is already tested on cypress, so maybe for this case is not necessary to test it again in Jasmine. I would like to know if you agreed.

Reply:
yeah this test is not needed. Especially as you are not using the routerLink. In your case when you click home it will reload the app vanilla way, whereas routerLink will not.